core/src/commands/workflow.ts

```diff
@@ -186,6 +187,7 @@ export class WorkflowCommand extends Command<Args, {}> {
         stepErrors[index] = [err]
         printStepDuration({ ...stepParams, success: false })
         logErrors(stepBodyLog, [err], index, steps.length, step.description)
+        shouldError = shouldError || !step.allowFailure
```

Comment:
It is more verbose, but I'd find this slightly more readable:
```suggestion
        if (!step.allowFailure) {
            shouldError = true
        }
```